### ISSUE TITLE
be resilient to decoding issues?

### ISSUE THREAD

**Issue by yarikoptic** [akkana/scripts#15]
```
No filename; making one up                                                                                                                                                                                                         
/home/yoh/.tmp/tmprz_ti1ww/part-001.txt doesn't have a Content-Id, not saving
No filename; making one up
/home/yoh/.tmp/tmprz_ti1ww/part-001.html doesn't have a Content-Id, not saving
Traceback (most recent call last):
  File "/home/yoh/.mutt/soft/viewhtmlmail", line 183, in <module>
    view_html_message(stdin, tmpdir)
  File "/home/yoh/.mutt/soft/viewhtmlmail", line 150, in view_html_message
    htmlsrc = html_part.get_payload(decode=True).decode('utf-8')
UnicodeDecodeError: 'utf-8' codec can't decode byte 0xe9 in position 28462: invalid continuation byte
```

not sure if that email doesn't have any sensitive information, so will now send it via email to the address in git commits.

PS THANK YOU for the viewhtml -- I had been using it for years!

**Comment by akkana** [comment]
Thanks for the test case that made this easy to reproduce. I've changed it to use the U+FFFD, REPLACEMENT CHARACTER, using python's "replace" errors argument. I also tried "backslashreplace" but they look the same to me in a browser at least for this test case: in both cases the character displays as "\u2013".

U+2013 is an en dash in UTF-8 according to https://www.utf8-chartable.de/unicode-utf8-table.pl -- seems like that should work, and I don't know why Python gives an error. I'm going to go through this code more carefully, and will see if I can figure out how to get it to display an endash instead of u2013. But at least it won't raise an error any more.

**Comment by akkana** [comment]
I think I see now. U+2013 is the unicode code point for that endash, but the UTF-8 code for it is e2 80 93. So since the email message claims it uses UTF-8 encoding, it should use the three byte code, and it's an error that it has the 2013 code in it instead.

I found that calling html_part.get_payload() instead of html_part.get_payload(decode=True).decode('utf-8') will show the endash -- but that breaks all the legitimate messages I tried, so I think it's more correct to show it with the \u code when there's an incorrect character in a message.
